cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/AdaptTemplatesCommand.java

```diff
@@ -0,0 +1,239 @@
+                    createFile(file, entry, saveForFileCreationPath);
+                }
+            }
+        } catch (IOException e) {
+            throw new IOException("An exception occurred while unpacking classes from Jar files to templates folder");
+        }
+    }
+
+    /**
+     * Creates a file or directory and cleans up current file if it exists already
+     * @param file
+     *            ZipFile to access
+     * @param entry
+     *            ZipEntry to get input stream from
+     * @param saveForFileCreationPath
+     *            Path to save file at
+     * @throws IOException
+     *             if file could not be created
+     */
+    private static void createFile(ZipFile file, ZipEntry entry, Path saveForFileCreationPath) throws IOException {
+        if (entry.isDirectory()) {
+            Files.createDirectories(saveForFileCreationPath);
+        } else {
+            Files.deleteIfExists(saveForFileCreationPath);
+            Files.createFile(saveForFileCreationPath);
+            try (InputStream is = file.getInputStream(entry);
+                BufferedInputStream bis = new BufferedInputStream(is);
+                FileOutputStream fileOutput = new FileOutputStream(saveForFileCreationPath.toString());) {
+
+                while (bis.available() > 0) {
+                    fileOutput.write(bis.read());
+                }
+            } catch (IOException e) {
+                throw new RuntimeIoException(e, IoMode.WRITE);
+            }
+        }
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        if (verbose) {
+            CLILogger.setLevel(Level.DEBUG);
+        }
+
+        logger.info("Running AdaptTemplatesCommand!, please wait...");
+
+        Path cobigenTemplatesDirectory = null;
+        if (customTemplatesLocation != null) {
+            customTemplatesLocation = configurationUtils.preprocessInputFile(customTemplatesLocation);
+            configurationUtils.createConfigFile(customTemplatesLocation);
+            // sets custom templates directory path from configuration file property
+            cobigenTemplatesDirectory = configurationUtils.getCustomTemplatesLocation();
+            logger.info("Creating custom templates folder at custom location @ {}", cobigenTemplatesDirectory);
+        } else {
+            // sets default templates directory path from CLI location
+            cobigenTemplatesDirectory = configurationUtils.getCobigenCliRootPath();
+            configurationUtils.createConfigFile(cobigenTemplatesDirectory.toFile());
+            logger.info("Creating custom templates folder next to the CLI @ {}", cobigenTemplatesDirectory);
+        }
+
+        if (Files.exists(cobigenTemplatesDirectory)) {
+            processJar(cobigenTemplatesDirectory);
+        } else {
+            logger.error("Could not find target directory to extract templates @", cobigenTemplatesDirectory);
```

Comment:
parameter is not included into the error message

Reply:
adjusted.